packages/contracts/scripts/generate-artifacts.ts

```diff
@@ -0,0 +1,62 @@
+import path from 'path'
+import glob from 'glob'
+import fs from 'fs'
+
+/**
+ * Script for automatically generating a file which has a series of `require` statements for
+ * importing JSON contract artifacts. We do this to preserve browser compatibility.
```

Comment:
This relies on `fs` so unsure what you mean by this statement of preserving browser compatibility

Reply:
This script generates a file and the generated file is designed to preserve browser compat